Sources/FoilDefaultStorage.swift

```diff
@@ -55,7 +55,7 @@ public struct FoilDefaultStorage<T: UserDefaultsSerializable> {
         // and uses force unwrap
         self._publisher = CurrentValueSubject<T, Never>(userDefaults.fetch(keyName))
 
-        self._observer = ObserverTrampoline(userDefaults: userDefaults, key: keyName) { [unowned _publisher] in
+        self._observer = ObserverTrampoline(userDefaults: userDefaults, key: keyName) { [_publisher] in
```

Comment:
> A strong reference doesn't cause a cycle, so I chose it over the theoretically slower weak reference.

Are we sure about this? I think this _does_ cause a cycle, right?

`self` --> `self._observer` --> `action` (i.e., `self._observer.action`) --> `_publisher` (i.e., `self._publisher`)

The capture of `_publisher` implicitly captures `self`, so we have:

`self` --> `observer` --> `action` --> `self`

Reply:
I wondered about that too, but it seems that the capture list `[_publisher]` expands to `[_publisher = self._publisher]` and does not capture `self`. I verified with the memory graph debugger that the CurrentValueSubject and ObserverTrampoline both disappear after I clear out an array of `FoilDefaultStorage`s.

I'm happy to use the more explicit syntax, I got the same results either way with the memory graph debugger.

Reply:
Ohh I see.

I think I'd prefer to just capture `[weak self]` which feels much more clear and obvious to me, and I don't think we need to worry about any potential performance impact.

Reply:
`self` is a `struct` in that location so I can't do `[weak self]`. How about `[weak _publisher]`?

Reply:
Oh wait. Using `weak self` probably presents issues with Swift initialization rules, right?

If so, let's do this:

```swift
let publisher = // ....
self._publisher = publisher
self._observer = ObserverTrampoline(userDefaults: userDefaults, key: keyName) {
    publisher.send(userDefaults.fetch(keyName))
}
```

Reply:
Or, alternately, create the publisher with `let publisher = CurrentValueSubject(…)`, set `_publisher = publisher`, then I can use `publisher` freely in the closure. I think I prefer this one.

Reply:
I pushed that last alternative, let me know if it's clear enough :)

Reply:
Ha perfect 😄